datafusion/src/datasource/parquet.rs

```diff
@@ -38,11 +38,22 @@ pub struct ParquetTable {
     schema: SchemaRef,
     statistics: Statistics,
     max_concurrency: usize,
+    enable_pruning: bool,
 }
 
 impl ParquetTable {
     /// Attempt to initialize a new `ParquetTable` from a file path.
     pub fn try_new(path: impl Into<String>, max_concurrency: usize) -> Result<Self> {
+        ParquetTable::try_new_with_pruning_config(path, max_concurrency, true)
+    }
+
+    /// Attempt to initialize a new `ParquetTable` from a file path. And enable or
+    /// disable the parquet pruning features.
+    pub fn try_new_with_pruning_config(
```

Comment:
Here I'm not sure if adding the function is a good choice. 

My concern is, it is a public function, there may be many users who rely on it. But the `enable_pruning` in the signature is somehow temporal, we don't want it to last for a long time.

So I have another thought, replace this function with `try_new_with_config(path: impl Into<String>,, execution_config: ExecutionConfig)`. I think it's a better option, but it will introduce the dependency of `execution::context` module which I think is the top-level module. It seems a little weird. 

I'm not sure if the second method is acceptable?

Reply:
how about adopting the same builder design you went with `pub fn with_parquet_pruning`? then you can initialize it as:

```rust
            ParquetTable::try_new(
                filename,
                m.config.concurrency,
            )?.with_enable_pruning(m.config.parquet_pruning)
```

Reply:
@houqp Thank you for your advice. I think it is more elegant than my first version, and I've modified it to this style.